### DISCUSSION TITLE
How is Laravel a MIT project if it uses a GPL libraries like lib-array2xml

### DISCUSSION THREAD

**Discussion by ProteanCode** [laravel/framework#34883] [Q&A] [answered]
It uses a  [lib-array2xml](https://github.com/nullivex/lib-array2xml) library which is GPL v3

**Comment by GrahamCampbell** [comment, marked as answer]
Laravel itself is, but that doesn't mean all its dependencies are.

**Reply by ProteanCode** [reply]
But this does not work that way. I mean you can release the framework as a MIT, sure, but you inherit all GPL flaws when you include such dependency right?

**Reply by ProteanCode** [reply]
> Laravel itself is, but that doesn't mean all its dependencies are.

Let's say that Laravel Forge includes the GPL library, how the provisioner implementation stays closed source then?

**Reply by GrahamCampbell** [reply]
GPL allows for closed source server software. Only AGPL does not.
